polyfill/lib/duration.mjs

```diff
@@ -129,15 +129,18 @@ export class Duration {
     if (GetSlot(this, MINUTES)) timeParts.push(`${GetSlot(this, MINUTES)}M`);
 
     const secondParts = [];
-    if (GetSlot(this, NANOSECONDS)) secondParts.unshift(`000${GetSlot(this, NANOSECONDS)}`.slice(-3));
-    if (GetSlot(this, MICROSECONDS) || secondParts.length) {
-      secondParts.unshift(`000${GetSlot(this, MICROSECONDS)}`.slice(-3));
-    }
-    if (GetSlot(this, MILLISECONDS) || secondParts.length) {
-      secondParts.unshift(`000${GetSlot(this, MILLISECONDS)}`.slice(-3));
-    }
+    let s = GetSlot(this, SECONDS);
+    let ms = GetSlot(this, MILLISECONDS);
+    let µs = GetSlot(this, MICROSECONDS);
+    let ns = GetSlot(this, NANOSECONDS);
+    let deltaSeconds;
+    ({ deltaSeconds, millisecond: ms, microsecond: µs, nanosecond: ns } = ES.BalanceSubSecond(ms, µs, ns));
+    s += deltaSeconds;
```

Comment:
Nit: could be `const s = GetSlot(this, SECONDS) + deltaSeconds`, rather than initializing in two steps. Might be clearer.